### DISCUSSION TITLE
What should we add next to the Dagger Universe?

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#1922] [General]
An important feature of Dagger is [Universe](https://github.com/dagger/dagger/tree/main/pkg/universe.dagger.io): a curated collection of reusable actions, which any Dagger developer can add into their plan.

So far we have actions for [Docker](https://docker.com), [Netlify](https://netlify.com), [AWS](https://aws.amazon.com), [Yarn](https://yarnpkg.com), just to name a few.

Anyone can develop their own actions, but it's important to expand the collection available in Universe, so that developers don't have to re-invent the wheel every time.

So the question is... ***which actions should we add to Dagger Universe next?***

Make suggestions and vote below :)

**Comment by shykes** [comment]
One common request is to add actions for `kubernetes`:

* Generate a Kubernetes config
* Validate a config
* Apply a config with `kubectl apply`
* Support native authentication to popular kubernetes hosting providers (AWS, Azure, Google Cloud, Alicloud, etc)
* Perhaps integrate with popular companion tools, such as kustomize?

**Reply by snowmead** [reply]
I would extend this request further to add actions for `helm`.
What do you think?

**Comment by arnaud-soulie** [comment]
`vault` client:
- Read and write secrets
- Generate and renew tokens
- ...

**Comment by snowmead** [comment]
Add an action for `terraform`:
- init Backend remote state
- plan
- apply
- workspaces
- ...

**Comment by mtumalewicz** [comment]
Wdyt about notification/notify module for handling authorization and sending messages via the most popular channels/apps:
* mail
* slack
* mattermost
* teams
etc.

**Comment by theBeginner86** [comment]
Using [Meshery](https://github.com/meshery/meshery) for
- service mesh provisioning
- canary performance testing

**Comment by vdemeester** [comment]
Probably a bit generic but adding more language "build" support. Today there is *bash*, `go`, `python` and *yarn* (JS right ?). We could add more things here `ruby`, `java`, `clojure`, `rust`, …

**Reply by eddumelendez** [reply]
java:
- maven
- gradle

clojure:
- leiningen
- clojure-tools

for the first 3 (maven, gradle, leiningen), those can be also installed via [sdkman](https://sdkman.io/sdks) and use `.sdkmanrc` file to install multiples sdks and setup and environment. See https://sdkman.io/usage#env so, sdkman action would also be great.

**Comment by gerhard** [comment]
[Vercel](https://vercel.com/)

cc @nabsul @marcosnils

**Comment by gerhard** [comment]
[Fly.io](https://fly.io/)

This is my starting point here: https://github.com/thechangelog/changelog.com/pull/407/files#diff-7f9a218d671a3accf380cc0a32121c588ecec518e18dabdb544b83f3f64fb8a9

cc @mrkurt @brainlid @chrismccord

**Comment by gerhard** [comment]
[Fastly](https://www.fastly.com/)

I would especially like to [move this](https://github.com/thechangelog/changelog.com/blob/7b8c2a48a01fc11aa2c5848c680c589a09db0b66/2021/manifests/changelog/jobs.yml#L18-L72) into a Dagger plan.

**Comment by gerhard** [comment]
[LastPass](https://www.lastpass.com/)

[There is a lot of that here](https://github.com/thechangelog/changelog.com/blob/7c98c6818d014e13572b3911d7e854080fa44d4e/2021/secrets.mk), and while we intend to move to something like Vault, we are still using LastPass today.

**Comment by berryp** [comment]
[1Password](https://1password.com/developers/) and [1Password Secrets Automation](https://1password.com/products/secrets/)

**Comment by nmeisenzahl** [comment]
**Support for Azure**

Secret management: Retrieving secrets from Azure Key Vault

Deployments: Deploying to Azure Static Web Apps, Azure Functions, Azure Container Apps, ...

IaC: Bicep

IMO those would be a good starting point.

**Comment by schosterbarak** [comment]
IaC security scanning: using [checkov](https://github.com/bridgecrewio/checkov/)

**Comment by cvantum** [comment]
For password storage i really would like to see passwordstore (https://www.passwordstore.org/)
